python/gstaichi/lang/kernel_impl.py

```diff
@@ -713,6 +713,10 @@ def materialize(self, key: CompiledKernelKeyType | None, args: tuple[Any, ...],
                 )
                 if self.compiled_kernel_data_by_key[key]:
                     self.src_ll_cache_observations.cache_loaded = True
+        elif self.gstaichi_callable and not self.gstaichi_callable.is_pure and self.runtime.print_non_pure:
+            # bit in caps should not be modified without updating corresponding test
+            # freetext can be freely modified
+            print(f"[NOT_PURE] Not pure: {self.func.__name__}")
```

Comment:
As a person doesn't know taichi, the printed information has no meaning to me. is it an error or a warning?